### PR TITLE
WIP Swarm/Consul

### PR DESCRIPTION
**Pull request by till**
For: pngmbh/issues#496

### CODE REVIEW

**Comment by rwos** [comment]
re: traefik test: it's using `Host:traefik.traefik-swarm.docker.local` but molecule/default/playbook.yml has `traefik_docker_domain: 'traefik.docker.local'`, not `traefik-swarm.docker.local`

**Comment by rwos** [comment]
the storeconfig doesn't seem to work on circleci, sometimes(?): from the traefik_init container:
```
2020/02/26 17:28:27 Error running traefik: Put http://consul:8500/v1/kv/traefik/accesslogsfile?flags=3304740253564472344&wait=30000ms: dial tcp 10.0.2.2:8500: connect: no route to host
```

**Comment by rwos** [comment]
So, it's nothing wrong with consul, I think somehow the whole docker network stuff doesn't work - can't ping or connect to any container from any other one (but the hostnames all resolve).

No idea where in the layers of docker-swarm in docker in circleci it all goes wrong, but probably somewhere.

**Comment by rwos** [comment]
now the `pip install docker` fails, for some reason. So much fun.

**Comment by till** [comment]
Oh, I just remembered. I think we have to configure a network explicitly in Molecule. No idea. I remember wasting a few hours on that before.

**Comment by rwos** [comment]
I switched this over to hetzner now, so that *should* work now. But haven't looked through the pip error messages yet. It's a bit weird that this doesn't work out of the box, because the same stuff works in other repos...

**Comment by rwos** [comment]
aaaand it's back at the same error. Now that was useful 🙈

File: molecule/swarm/playbook.yml [hunk 1]
```diff
@@ -4,20 +4,48 @@
...
+      ignore_errors: true
+      failed_when: false
+      tags:
+        - skip_ansible_lint
+    - name: Restart traefik
+      command:
+        cmd: docker service update --force traefik_traefik
+    - name: Wait for traefik/consul to come up
+      wait_for:
+        port: 80
```
Comment (rwos) [inline molecule/swarm/playbook.yml:76]: So this works. It's also horrible. From hacking around it seems one needs to do one request and then restart traefik before everything works properly. I have no idea why. Sometimes, just a restart is enough - but what never seems to work is just letting it all come up cleanly from scratch.

*Maybe* things would work better if traefik would actually wait until the consul is running and configured before it starts up, but not 100% sure if that is the core of the issue.

File: templates/docker-compose.yml.j2 [hunk 2]
```diff
@@ -5,101 +5,33 @@ services:
...
-    image: consul:1.6
-    command:
-      - "agent"
-      - "-server"
-      - "-ui"
-      - "-node=traefik-consul"
-      - "-bootstrap-expect=1"
-      - "-log-level=debug"
-      - "-client=0.0.0.0"
+      - "--configfile=/traefik.toml"
```
Comment (rwos) [inline templates/docker-compose.yml.j2:19]: aahh 😄 👍

**Review by rwos — APPROVED** [review]
LGTM

**Pull request merged**